modules/datepicker/basepicker.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+import {Button} from 'react-native'
+
+import {useTheme} from '@frogpond/app-theme'
+import type {AppTheme} from '@frogpond/app-theme'
+
+import moment from 'moment-timezone'
+import type {Moment} from 'moment-timezone'
+
+import DateTimePicker from '@react-native-community/datetimepicker'
+import type {Event} from '@react-native-community/datetimepicker'
+
+import {BaseDatetimePickerProps} from './types'
+
+const FORMATS = {
+	date: 'YYYY-MM-DD',
+	datetime: 'YYYY-MM-DD HH:mm',
+	time: 'HH:mm',
+}
+
+export const BaseDateTimePicker = (
```

Comment:
More of just a curiosity: any particular reason we aren't doing a `class` or components or something?

Reply:
Good question! This is being turned into a functional component because we can strike with a lighter hammer here. This could remain a class, for sure, but I applied a personal fule when refactoring this: I asked "if this can be done via a functional component, make it one". 

Hooks also offer a lot of niceties if we decide to optimize any bit of this in the future (e.g. useEffect, useCallback, useMemo) and allows us to improve readability by not falling into adding more old-style component lifecycle methods to accomplish the same ideas.

You did inspire another thought! I removed the `ref` which added a tappable callback to the row. It should be added back here via `useRef`.